pkg/kubelet/pod_workers.go

```diff
@@ -820,7 +820,8 @@ func (p *podWorkers) allowStaticPodStart(fullname string, uid types.UID) bool {
 	waitingPods := p.waitingToStartStaticPodsByFullname[fullname]
 	for i, waitingUID := range waitingPods {
 		// has pod already terminated or been deleted?
-		if _, ok := p.podSyncStatuses[waitingUID]; !ok {
+		status, ok := p.podSyncStatuses[waitingUID]
+		if !ok || status.IsTerminationRequested() || status.IsTerminated() {
```

Comment:
This condition is not correct - we can't start the pod until it is fully and absolutely removed.  How did we end up with the pod still in this sync status?

Note that cleanup takes a significant period of time in some workloads, so with this change it could be tens of seconds after the pod finishes termination before it's fully cleaned up.  Is that being taken into account?

Reply:
Ok, after looking at the e2e run I see the same thing as you.  The pod was added to waiting, then terminated, but the pod worker still has to run.

I like the cleanliness of keeping the logic here, but this does expose us to the potential of a large number of updates being kept (contiuously updated static pod would generate O(updates) worth of map entries, instead of O(1)).  Let me look a bit further at what we could do here.

Reply:
I'd say we stay focused now without changing where this code lives - you can abuse kubelet worse simply by creating lots of pods.

Reply:
Mabye we just leave a TODO here which is "This is O(N) with respect to number of updates to static pods with overlapping names, and ideally would be O(1)"

Reply:
done